clang/include/clang/Basic/Attr.td

```diff
@@ -1174,6 +1174,22 @@ static llvm::Triple::EnvironmentType getEnvironmentType(llvm::StringRef Environm
   let Documentation = [AvailabilityDocs];
 }
 
+def DomainAvailability : InheritableAttr {
+  let Spellings = [Clang<"domain_availability">];
+  let Args = [StringArgument<"domain">, BoolArgument<"unavailable">];
+  let AdditionalMembers = [{
+    std::string getFeatureAttributeStr() const;
+  }];
+  let Documentation = [Undocumented];
+}
+
+def AvailabilityDomain : InheritableAttr {
```

Comment:
Some comments and/or documentation on both of these would be good, since they have names that suggest they mean the same thing.